### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by m4dc4p**
Make it clear that this shell requires cloning the GHC repo first.

Based on my experience in #106.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 Simple usage
 ============
 
+## First Things First
```
Comment (supersven) [inline README.md:4]: Hey @m4dc4p ,

Thanks a lot for this PR! Lowering the entry barrier for newcomers is indeed a very good goal.

However, I'm not convinced that this section transports the information as good as it could. I'd like to propose to replace it by a short introduction at the top of this file that covers:
- This nix environment is meant for GHC development.
- You need a cloned GHC repo.
- Link to https://ghc.dev

So, basically what you already have, but reverb `This shell is intended to run in the context of the GHC repo` and move it to the top.

Would be great if you could find some time to do this. Otherwise, I can.

**Review by supersven — CHANGES_REQUESTED** [review]

**Comment by m4dc4p** [comment]
Feel free - thanks for looking!

**Comment by supersven** [comment]
@m4dc4p ,

> Feel free - thanks for looking!

Okay, please take a look at this PR: https://github.com/alpmestan/ghc.nix/pull/119
